libsweep/serial.h

```diff
@@ -7,24 +7,29 @@
  */
 
 #include "sweep.h"
+#include "Error.h"
 
-#include <stdint.h>
+#include <memory>
 
 namespace sweep {
 namespace serial {
 
-typedef struct device* device_s;
-typedef struct error* error_s;
-
-const char* error_message(error_s error);
-void error_destruct(error_s error);
-
-device_s device_construct(const char* port, int32_t bitrate, error_s* error);
-void device_destruct(device_s serial);
-
-void device_read(device_s serial, void* to, int32_t len, error_s* error);
-void device_write(device_s serial, const void* from, int32_t len, error_s* error);
-void device_flush(device_s serial, error_s* error);
+struct Error : public ErrorBase {
+	using ErrorBase::ErrorBase;
+};
+
+class Device {
+public:
+	Device(const char* port, int32_t bitrate);
+	Device(Device&& other) = default;
```

Comment:
Why a move ctor but not a move assignment operator? Also why not noexcept?

Let's go a step back here: why the unique_ptr state below anyway?